File: dotnet/In3/Btc/BlockHeader.cs
using System.Text.Json.Serialization;

namespace In3.Btc
{
    /// <summary>
    /// A Block header.
    /// </summary>
    public class BlockHeader
    {
        /// <summary>
        /// The hash of the blockheader.
        /// </summary>
        [JsonPropertyName("hash")] public string Hash { get; set; }

        /// <summary>
        /// Number of confirmations or blocks mined on top of the containing block.
        /// </summary>
        [JsonPropertyName("confirmations")] public uint Confirmations { get; set; }

        /// <summary>
        /// Block number.
        /// </summary>
        [JsonPropertyName("height")] public int Height { get; set; }

        /// <summary>
        /// Used version.
        /// </summary>
        [JsonPropertyName("version")] public int Version { get; set; }

        /// <summary>
        /// Version as hex.
        /// </summary>
        [JsonPropertyName("versionHex")] public string VersionHex { get; set; }

        /// <summary>
        /// Merkle root of the trie of all transactions in the block.
        /// </summary>
        [JsonPropertyName("merkleroot")] public string Merkleroot { get; set; }

        /// <summary>
        /// Unix timestamp in seconds since 1970.
        /// </summary>
        [JsonPropertyName("time")] public uint Time { get; set; }

        /// <summary>
        /// Unix timestamp in seconds since 1970.
        /// </summary>
        [JsonPropertyName("mediantime")] public uint Mediantime { get; set; }

        /// <summary>
        /// Nonce-field of the block.
        /// </summary>
        [JsonPropertyName("nonce")] public uint Nonce { get; set; }

        /// <summary>
        /// Bits (target) for the block as hex.
        /// </summary>
        [JsonPropertyName("bits")] public string Bits { get; set; }

        /// <summary>
        /// Difficulty of the block.
        /// </summary>
        [JsonPropertyName("difficulty")] public float Difficulty { get; set; }

        /// <summary>
        /// Total amount of work since genesis.
        /// </summary>
        [JsonPropertyName("chainwork")] public string Chainwork { get; set; }

        /// <summary>
        /// Number of transactions in the block.
        /// </summary>
        [JsonPropertyName("nTx")] public long NTx { get; set; }

        /// <summary>
        /// Hash of the parent blockheader.
        /// </summary>
        [JsonPropertyName("previousblockhash")] public string Previousblockhash { get; set; }

        /// <summary>
        /// Hash of the next blockheader.
        /// </summary>
        [JsonPropertyName("nextblockhash")] public string Nextblockhash { get; set; }
    }
}

File: java/src/in3/btc/BlockHeader.java
package in3.btc;

import in3.utils.JSON;

/**
 * A Block header.
 */
public class BlockHeader {

  private JSON data;

  protected BlockHeader(JSON data) {
    this.data = data;
  }

  protected static BlockHeader asBlockHeader(Object o) {
    if (o == null)
      return null;
    return new BlockHeader((JSON) o);
  }

  protected static BlockHeader[] asBlockHeaders(Object o) {
    if (o == null)
      return null;
    Object[] a      = (Object[]) o;
    BlockHeader[] b = new BlockHeader[a.length];
    for (int i = 0; i < a.length; i++)
      b[i] = BlockHeader.asBlockHeader(a[i]);
    return b;
  }

  /**
   * The hash of the blockheader.
   */
  public String getHash() {
    return data.getString("hash");
  }

  /**
   * Number of confirmations or blocks mined on top of the containing block.
   */
  public long getConfirmations() {
    return data.getLong("confirmations");
  }

  /**
   * Block number.
   */
  public long getHeight() {
    return data.getLong("height");
  }

  /**
   * Used version.
   */
  public long getVersion() {
    return data.getLong("version");
  }

  /**
   * Version as hex.
   */
  public String getVersionHex() {
    return data.getString("versionHex");
  }

  /**
   * Merkle root of the trie of all transactions in the block.
   */
  public String getMerkleroot() {
    return data.getString("merkleroot");
  }

  /**
   * Unix timestamp in seconds since 1970.
   */
  public long getTime() {
    return data.getLong("time");
  }

  /**
   * Unix timestamp in seconds since 1970.
   */
  public long getMediantime() {
    return data.getLong("mediantime");
  }

  /**
   * Nonce-field of the block.
   */
  public long getNonce() {
    return data.getLong("nonce");
  }

  /**
   * Bits (target) for the block as hex.
   */
  public String getBits() {
    return data.getString("bits");
  }

  /**
   * Difficulty of the block.
   */
  public float getDifficulty() {
    String floatStr = data.getString("difficulty");
    return Float.parseFloat(floatStr);
  }

  /**
   * Total amount of work since genesis.
   */
  public String getChainwork() {
    return data.getString("chainwork");
  }

  /**
   * Number of transactions in the block.
   */
  public long getNTx() {
    return data.getLong("nTx");
  }

  /**
   * Hash of the parent blockheader.
   */
  public String getPreviousblockhash() {
    return data.getString("previousblockhash");
  }

  /**
   * Hash of the next blockheader.
   */
  public String getNextblockhash() {
    return data.getString("nextblockhash");
  }
}
